frontend/src/components/LoginForm.jsx

```diff
@@ -1,10 +1,15 @@
 import { useRef } from "react";
+import { useNavigate } from "react-router-dom";
 import { ToastContainer, toast } from "react-toastify";
+import { useUserData } from "../context/UserDataContext";
 import "react-toastify/dist/ReactToastify.css";
 
 export default function LoginForm() {
+  const { setUserData } = useUserData([]);
```

Comment:
userData will be filled with an object : avoid passing an empty array as initial value ;)